pkg/util/kubeobjects/statefulset/query.go

```diff
@@ -32,10 +32,33 @@ func Query(kubeClient client.Client, kubeReader client.Reader, log logd.Logger)
 	}
 }
 
+func RecreateQuery(kubeClient client.Client, kubeReader client.Reader, log logd.Logger, customMustRecreate MustRecreateFunc) query.Generic[*appsv1.StatefulSet, *appsv1.StatefulSetList] {
+	return query.Generic[*appsv1.StatefulSet, *appsv1.StatefulSetList]{
```

Comment:
There's lot's of duplicate code. Maybe this can go into a common function that accepts a "recreate" function.

```
func query(kubeClient client.Client, kubeReader client.Reader, log logd.Logger, mustRecreate MustRecreateFunc) ... {
  return query.Generic[*appsv1.StatefulSet, *appsv1.StatefulSetList]{ 
    ... 
    MustRecreate: mustRecreate
    ...
  }
}

func RecreateQuery(...) ... {
  recreateFunc := func(current, desired *appsv1.StatefulSet) bool {
			return mustRecreate(current, desired) || customMustRecreate(current, desired)
		}
  return query(..., recreateFunc)
}

func Query(...) ... {
  return query(..., mustRecreate)
}
```